chain/network/src/snapshot_hosts/mod.rs

```diff
@@ -57,14 +182,59 @@ impl Inner {
         self.hosts.push(d.peer_id.clone(), d.clone());
         Some(d)
     }
+
+    // Try to insert up to max_entries_added more peers into the state part selector for this part ID
+    fn insert_part_hosts(
+        &mut self,
+        sync_hash: &CryptoHash,
+        shard_id: ShardId,
+        part_id: &PartId,
+        max_entries_added: usize,
+    ) {
+        let selector = self.state_part_selectors.get(&shard_id).unwrap();
+        let seen_peers = selector.seen_peers(part_id);
+
+        let mut new_peers = BTreeSet::new();
```

Comment:
This can also be a heap.

Reply:
Oh yea at first I thought it wasn't possible because we're looking for the top N, and want to compare with the _worst_ one of the N we've seen each time. But yeah we can do it by just flipping the comparator, good point